ZenPacks/zenoss/Microsoft/Windows/datasources/ServiceDataSource.py

```diff
@@ -208,50 +201,76 @@ def collect(self, config):
+                                                  'alertifnot': ds.params['alertifnot']}
+        return services
+
     def onSuccess(self, results, config):
 
         data = self.new_data()
-        ds0 = config.datasources[0]
-        serviceinfo = results[results.keys()[0]]
-        eventClass = ds0.eventClass if ds0.eventClass else "/Status"
-        eventKey = ds0.eventKey if ds0.eventKey else "WindowsService"
-
-        if serviceinfo[0].State != ds0.params['alertifnot']:
-
-            evtmsg = 'Service Alert: {0} has changed to {1} state'.format(
-                serviceinfo[0].Name,
-                serviceinfo[0].State
-            )
-
+        services = self.buildServicesDict(config.datasources)
+        log.debug('Windows services query results: {}'.format(results))
+        try:
+            serviceinfo = results[results.keys()[0]]
+        except:
             data['events'].append({
-                'eventClass': eventClass,
-                'eventClassKey': 'WindowsServiceLog',
-                'eventKey': eventKey,
-                'severity': ds0.severity,
-                'summary': evtmsg,
-                'component': prepId(serviceinfo[0].Name),
-                'device': config.id,
-            })
-        else:
-
-            evtmsg = 'Service Recoverd: {0} has changed to {1} state'.format(
-                serviceinfo[0].Name,
-                serviceinfo[0].State
-            )
-
-            data['events'].append({
-                'eventClass': eventClass,
-                'eventClassKey': 'WindowsServiceLog',
-                'eventKey': eventKey,
-                'severity': ZenEventClasses.Clear,
-                'summary': evtmsg,
-                'component': prepId(serviceinfo[0].Name),
-                'device': config.id,
-            })
+                                'eventClass': "/Status",
+                                'severity': ZenEventClasses.Warning,
+                                'eventClassKey': 'WindowsServiceCollectionError',
+                                'eventKey': 'WindowsServiceCollection',
+                                'summary': 'No results returned for service query',
+                                'device': config.id})
+            return data
+
+        for index in range(0, len(serviceinfo)):
+            if serviceinfo[index].State not in services.keys():
+                continue
+
+            service = services[serviceinfo[index].Name]
+            eventClass = config.datasources[index].eventClass if service['eventClass'] else "/Status"
+            eventKey = service['eventKey'] if service['eventKey'] else "WindowsService"
+
+            if serviceinfo[index].State != service['alertifnot']:
+
+                evtmsg = 'Service Alert: {0} has changed to {1} state'.format(
+                    serviceinfo[index].Name,
+                    serviceinfo[index].State
+                )
+
+                data['events'].append({
```

Comment:
It would be useful to add additional details from serviceinfo to the event such as unaltered values for Name and State. These things are useful in transforms so you don't have to use a regex to try to parse them out of the summary.